README.markdown

```diff
@@ -6884,6 +6888,26 @@ See also [ngx.shared.DICT](#ngxshareddict).
 
 [Back to TOC](#nginx-api-for-lua)
 
+ngx.shared.DICT.cog
+-------------------
+**syntax:** *value, flags = ngx.shared.DICT:cog(key, old_value?, old_flags?)*
+
+**context:** *set_by_lua&#42;, rewrite_by_lua&#42;, access_by_lua&#42;, content_by_lua&#42;, header_filter_by_lua&#42;, body_filter_by_lua&#42;, log_by_lua&#42;, ngx.timer.&#42;, balancer_by_lua&#42;, ssl_certificate_by_lua&#42;, ssl_session_fetch_by_lua&#42;, ssl_session_store_by_lua&#42;, ssl_client_hello_by_lua&#42;*
+
+Similar to the [get](#ngxshareddictget) method, but only returns if
+`old_value` or `old_flags` do not match.
+
+If `old_value` or `old_flags` is `nil`
+it will be ignored when comparing.
```

Comment:
This semantics does not feel right to me. When old value is nil, then it means the expected value in the shdict should also be nil. It should not get ignored? So it always return `nil, nil`in this case?

Reply:
When old value is nil, it should be ignored when comparing (so value could be anything), when old flag is nil, flag is ignored.
This way the api user could have either old_value or old_flag (or both) and whichever is not nil will be compared to the value in shm